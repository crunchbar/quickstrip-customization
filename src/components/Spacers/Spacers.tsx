import React from 'react';
import {Droppable} from 'react-beautiful-dnd';
import {Paper, Typography} from '@material-ui/core';
import {
  COMMON_ITEM_CLASS,
  SPACERS_ID,
  SPACER_ITEM_CLASS,
  VISIBLE_SPACER_ID,
  SPACER_ID,
} from '../../constants/constants';
import { newVisibleSpacer, newSpacer } from '../../utils/utils';
import QuickstripItem from '../QuickstripItem/QuickstripItem';

const Spacers: React.FC = () => {
  const [visibleSpacer] = React.useState(newVisibleSpacer());
  const [spacer] = React.useState(newSpacer());
  const commonClassName = `${COMMON_ITEM_CLASS} ${SPACER_ITEM_CLASS}`;
  return (
    <Paper id={SPACERS_ID} tabIndex={0} className="spacers vertical-space-1 border">
      <Typography variant="h5" component="h1" className="pad-1">
        Spacers
      </Typography>
      <div className="spacer-container">
        <span>Spacer</span>
        <Droppable droppableId={SPACER_ID} isDropDisabled={true}>
          {(provided) => (
            <div
              id={SPACER_ID}
              ref={provided.innerRef}
              {...provided.droppableProps}>
              <QuickstripItem
                className={commonClassName}
                item={spacer}
                index={0}
              />
              {provided.placeholder}
            </div>
          )}
        </Droppable>
      </div>
      <div className="spacer-container">
        <span>Visible Spacer</span>
        <Droppable droppableId={VISIBLE_SPACER_ID} isDropDisabled={true}>
          {(provided) => (
            <div
              id={VISIBLE_SPACER_ID}
              ref={provided.innerRef}
              {...provided.droppableProps}>
              <QuickstripItem
                className={commonClassName}
                item={visibleSpacer}
                index={0}
              />
              {provided.placeholder}
            </div>
          )}
        </Droppable>
      </div>
    </Paper>
  );
}

export default Spacers;
