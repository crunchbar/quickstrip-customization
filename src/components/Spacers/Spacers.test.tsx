import * as React from 'react';
import {shallow} from 'enzyme';
import Spacers from './Spacers';

describe('<Spacers />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Spacers />);
    expect(wrapper).toMatchSnapshot();
  });
});
