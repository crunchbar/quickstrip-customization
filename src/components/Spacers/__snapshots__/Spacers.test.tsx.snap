// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacers /> renders correctly 1`] = `
<WithStyles(ForwardRef(Paper))
  className="spacers vertical-space-1 border"
  id="spacers"
  tabIndex={0}
>
  <WithStyles(ForwardRef(Typography))
    className="pad-1"
    component="h1"
    variant="h5"
  >
    Spacers
  </WithStyles(ForwardRef(Typography))>
  <div
    className="spacer-container"
  >
    <span>
      Spacer
    </span>
    <Connect(Droppable)
      direction="vertical"
      droppableId="separator"
      getContainerForClone={[Function]}
      ignoreContainerClipping={false}
      isCombineEnabled={false}
      isDropDisabled={true}
      mode="STANDARD"
      type="DEFAULT"
      whenDraggingClone={null}
    >
      <Component />
    </Connect(Droppable)>
  </div>
  <div
    className="spacer-container"
  >
    <span>
      Visible Spacer
    </span>
    <Connect(Droppable)
      direction="vertical"
      droppableId="separator-visible"
      getContainerForClone={[Function]}
      ignoreContainerClipping={false}
      isCombineEnabled={false}
      isDropDisabled={true}
      mode="STANDARD"
      type="DEFAULT"
      whenDraggingClone={null}
    >
      <Component />
    </Connect(Droppable)>
  </div>
</WithStyles(ForwardRef(Paper))>
`;
